core/app/models/spree/variant.rb

```diff
@@ -253,6 +253,10 @@ def discontinued?
       !!discontinue_on && discontinue_on <= Time.current
     end
 
+    def backordered?
+      total_on_hand <= 0 && stock_items.where(backorderable: true).any?
```

Comment:
@bbonislawski  `&& stock_items.exists?(backorderable: true)` - How about it?

Reply:
👍